### DISCUSSION TITLE
Low Data Mode

### DISCUSSION THREAD

**Discussion by Moshyfawn** [zed-industries/zed#23926] [Feature Requests]
### Check for existing issues

- [X] Completed

### Describe the feature

It would be nice to have a "Low Data Mode" _(iOS-specific [term](https://support.apple.com/en-us/102433))_ feature that would disable all automatic network requests. This could include:

- [ ] Prioritising local language server binaries over downloading/updating them _(Ref: #4978)_
- [ ] Checking for updates less frequently / not at all _(Ref: #11386)_
- [ ] Disabling assistant / copilot features that require network requests _(Fallback to local LLMs? Is that possible at this time? Ref: #4424)_.
- [ ] Disable chat collaboration / "go offline" _(Ref: #10355)_

There may be other features that could be disabled to save data, but these are the ones that come to mind.

This feature could provide a prominent UI element to toggle each disabled feature, so the user can easily access it if they need to.

**Comment by JosephTLyons** [comment]
It should probably disable telemetry too?

**Comment by Moshyfawn** [comment]
> It should probably disable telemetry too?

It'd be nice, yea

**Comment by mikayla-maki** [comment]
This is a good idea! We should do this :)
